addons/fleet/models/fleet_vehicle_log_contract.py

```diff
@@ -83,6 +83,21 @@ def _compute_days_left(self):
 
     def write(self, vals):
         res = super(FleetVehicleLogContract, self).write(vals)
+        if 'start_date' in vals or 'expiration_date' in vals:
+            date_today = fields.Date.from_string(fields.Date.today())
+            futur_contracts = self.env[self._name]
+            running_contracts = self.env[self._name]
+            expired_contracts = self.env[self._name]
+            for contract in self.filtered(lambda c: c.start_date and c.state != 'closed'):
+                if date_today < contract.start_date:
+                    futur_contracts |= contract
+                elif not contract.expiration_date or contract.start_date < date_today < contract.expiration_date:
+                    running_contracts |= contract
+                else:
+                    expired_contracts |= contract
+            futur_contracts.write({'state': 'futur'})
+            running_contracts.write({'state': 'open'})
+            expired_contracts.write({'state': 'expired'})
```

Comment:
We could reuse the action methods just below
```suggestion
            futur_contracts.write({'state': 'futur'})
            running_contracts.write({'state': 'open'})
            expired_contracts.write({'state': 'expired'})
```